### PR TITLE
Fixed the description of Half Cheetah's observation space elements

### LINKED ISSUE #906: [Bug Report] Bug in docs or bug in implementation of HalfCheetah Mujoco Env
### Describe the bug

In the docs page of Cheetah:   (https://gymnasium.farama.org/environments/mujoco/half_cheetah/)

We have:

**excluded | x-coordinate of the front tip | -Inf | Inf | rootx | slide | position (m)**

And:

**8 | x-coordinate of the front tip | -Inf | Inf | rootx | slide | velocity (m/s)**

https://github.com/Farama-Foundation/Gymnasium/blob/b8991ece32048af342bc82324bd94d5f8e3fa901/gymnasium/envs/mujoco/half_cheetah_v5.py#L56-L75

And in the code of the env, we have:

https://github.com/Farama-Foundation/Gymnasium/blob/b8991ece32048af342bc82324bd94d5f8e3fa901/gymnasium/envs/mujoco/half_cheetah_v5.py#L278-L281

And also:

https://github.com/Farama-Foundation/Gymnasium/blob/b8991ece32048af342bc82324bd94d5f8e3fa901/gymnasium/envs/mujoco/half_cheetah_v5.py#L251-L259

Following the Hopper and Walker envs it makes sense to assume that the right excluded *qpos* values is at index 0.

But then the docs are wrong.

Can you please clarify this?

Thanks

### Code example

_No response_

### System info

Gymnasium 1.0.0a

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by jamartinh**
… consistent with the xml file.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #906

## Type of change

Please delete options that are not relevant.

- [X ] Documentation only change (no code changed)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 1]
```diff
@@ -53,26 +53,28 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
...
-    | 12  | angle of the second rotor            | -Inf | Inf | bshin                            | hinge | angular velocity (rad/s) |
-    | 13  | velocity of the tip along the x-axis | -Inf | Inf | bfoot                            | hinge | angular velocity (rad/s) |
-    | 14  | velocity of the tip along the y-axis | -Inf | Inf | fthigh                           | hinge | angular velocity (rad/s) |
-    | 15  | angular velocity of front tip        | -Inf | Inf | fshin                            | hinge | angular velocity (rad/s) |
-    | 16  | angular velocity of second rotor     | -Inf | Inf | ffoot                            | hinge | angular velocity (rad/s) |
-    | excluded |  x-coordinate of the front tip  | -Inf | Inf | rootx                            | slide | position (m)             |
+
+    | Num | Observation                                 | Min  | Max | Name (XML) | Joint | Type (Unit)                |
+    | --- | ------------------------------------------- | ---- | --- | ---------- | ----- | -------------------------- |
+    | excluded | x-coordinate of the front tip          | -Inf | Inf | rootx      | slide | position (m)               |
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:59]: Excluded observation go to the bottom of list

**Review by Kallinteris-Andreas — COMMENTED** [review]
Sorry, for the miscommunication, the `Num` column should contain the index of the observation, not a mapping of state to observation

File: gymnasium/envs/mujoco/half_cheetah_v5.py (hunk 1 above)
Comment (jamartinh) [reply gymnasium/envs/mujoco/half_cheetah_v5.py:59]: Sorry but I don't understand what you mean by index vs mapping.
Also if you see the Hopper and walker, excluded is at first position.

Comment (Kallinteris-Andreas) [reply gymnasium/envs/mujoco/half_cheetah_v5.py:59]: 1) The Excluded observations are at the bottom, 
image

2) the index in `Num` is used to indicate the index of the observation, the observation with Num=0 should be in `observation[0]`, the observation with Num=1 should be in `observation[1]`, where `observation` is returned from `step` and `reset`

p.s. ignore the CI failing, so long as pre-commit passes it is fine, it fails for an unrelated to this PR problem

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 2]
```diff
@@ -136,6 +138,7 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
         - Added `frame_skip` argument, used to configure the `dt` (duration of `step()`), default varies by environment check environment documentation pages.
         - Restored the `xml_file` argument (was removed in `v4`).
         - Renamed `info["reward_run"]` to `info["reward_forward"]` to be consistent with the other environments.
+        - Fixed the observation space in doc string to be consistent with the xml file.
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:141]: This does not belong here, remove it. Documentation changes are listed in the gymnasium release notes

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 3]
```diff
@@ -53,26 +53,26 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
...
-    | 13  | velocity of the tip along the x-axis | -Inf | Inf | bfoot                            | hinge | angular velocity (rad/s) |
-    | 14  | velocity of the tip along the y-axis | -Inf | Inf | fthigh                           | hinge | angular velocity (rad/s) |
-    | 15  | angular velocity of front tip        | -Inf | Inf | fshin                            | hinge | angular velocity (rad/s) |
-    | 16  | angular velocity of second rotor     | -Inf | Inf | ffoot                            | hinge | angular velocity (rad/s) |
-    | excluded |  x-coordinate of the front tip  | -Inf | Inf | rootx                            | slide | position (m)             |
+    | Num | Observation                               | Min  | Max | Name (in corresponding XML file) | Joint | Type (Unit)              |
+    | --- | ----------------------------------------- | ---- | --- | -------------------------------- | ----- | ------------------------ |
+    | 0   | z-coordinate angle of the front tip       | -Inf | Inf | rootz                            | slide | position (m)             |
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:58]: This is not an angle

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 4]
```diff
@@ -53,26 +53,26 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
...
+    | 1   | angle of the root                         | -Inf | Inf | rooty                            | hinge | angle (rad)              |
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:59]: "front tip" is more descriptive (also change for velocity)

**Review by Kallinteris-Andreas — CHANGES_REQUESTED** [review]
A couple of changes, and we are good to merge

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 5]
```diff
@@ -53,26 +53,28 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
...
-    | 11  | angle of the second rotor            | -Inf | Inf | bthigh                           | hinge | angular velocity (rad/s) |
-    | 12  | angle of the second rotor            | -Inf | Inf | bshin                            | hinge | angular velocity (rad/s) |
-    | 13  | velocity of the tip along the x-axis | -Inf | Inf | bfoot                            | hinge | angular velocity (rad/s) |
-    | 14  | velocity of the tip along the y-axis | -Inf | Inf | fthigh                           | hinge | angular velocity (rad/s) |
-    | 15  | angular velocity of front tip        | -Inf | Inf | fshin                            | hinge | angular velocity (rad/s) |
-    | 16  | angular velocity of second rotor     | -Inf | Inf | ffoot                            | hinge | angular velocity (rad/s) |
-    | excluded |  x-coordinate of the front tip  | -Inf | Inf | rootx                            | slide | position (m)             |
+    I see, the issue is with the alignment of the table starting from the 8th row. The problem is due to the length of the text in the 'Observation' column. Here's how you can align it correctly:
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:56]: remove that,

File: gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 6]
```diff
@@ -53,26 +53,28 @@ class HalfCheetahEnv(MujocoEnv, utils.EzPickle):
...
+    | 3   | angle of the back shin                      | -Inf | Inf | bshin                            | hinge | angle (rad)              |
+    | 4   | angle of the back foot                      | -Inf | Inf | bfoot                            | hinge | angle (rad)              |
+    | 5   | angle of the front thigh                    | -Inf | Inf | fthigh                           | hinge | angle (rad)              |
+    | 6   | angle of the front shin                     | -Inf | Inf | fshin                            | hinge | angle (rad)              |
+    | 7   | angle of the front foot                     | -Inf | Inf | ffoot                            | hinge | angle (rad)              |
+    | 8   | velocity of the x-coordinate of front tip   | -Inf | Inf | rootx                            | slide | velocity (m/s)           |
+    | 9   | velocity of the z-coordinate of front tip   | -Inf | Inf | rootz                            | slide | velocity (m/s)           |
+    | 10  | angular velocity of the root                | -Inf | Inf | rooty                            | hinge | angular velocity (rad/s) |
```
Comment (Kallinteris-Andreas) [inline gymnasium/envs/mujoco/half_cheetah_v5.py:70]: "front tip"

File: gymnasium/envs/mujoco/half_cheetah_v5.py (hunk 5 above)
Comment (jamartinh) [reply gymnasium/envs/mujoco/half_cheetah_v5.py:56]: Copilot leak!

**Review by Kallinteris-Andreas — APPROVED** [review]
Thanks, for the work

**Pull request merged**